Improve update logging in Mongo data manager

peiyaoh opened the pull request:
Adds a warning log when an update operation matches an item but does not modify it. The server will still return the document, instead of null (so won't be an error). Refactors conditional logic for clarity in updateItemInCollection.

### Description

-


### Related Jira Tickets

-


### Comments 

-


Please ensure all items are complete before requesting a review:

- [ ] **Code Completed**: All code for the feature or bugfix has been implemented.
- [ ] **Unit Tests Added**: Unit tests are written for all new functionality, covering edge cases where applicable.
- [ ] **Integration Tests (where necessary)**: Integration tests are added for interactions between modules or components.
- [ ] **All Tests Passing**: All unit and integration tests are passing.
- [ ] **Type Safety Checked**: Ensure TypeScript types are correctly defined and used. Avoid `any` types without proper justification.
- [ ] **Typedoc Comments Added**: Ensure all functions, classes, interfaces, and types are documented using TypeDoc syntax.
- [ ] **Code Formatting**: All code files are formatted according to the project’s standards (e.g., Prettier, ESLint) and no linting errors remain.
- [ ] **No Console Logs**: Remove any `console.log` or other debugging output from the code.
- [ ] **Documentation Updated**: Update relevant project documentation (e.g., README, configuration guides) if necessary.

---

peiyaoh committed:
Improve update logging in Mongo data manager

Adds a warning log when an update operation matches an item but does not modify it. The server will still return the document, instead of null (so won't be an error). Refactors conditional logic for clarity in updateItemInCollection.

jadrubio commented on src/data-manager/mongo/mongo-data-manager.ts:
```diff
@@ -133,7 +133,7 @@ const getCollectionChangeReadable = (
 
   const collectionChangeReadable = new Readable({ objectMode: true });
-  collectionChangeReadable._read = () => {};
+  collectionChangeReadable._read = () => { };
```
no need for space

jadrubio commented on src/data-manager/mongo/mongo-data-manager.ts:
```diff
@@ -220,12 +220,17 @@ const updateItemInCollection = async (
+    if (matchedCount === 1) {
+      if (modifiedCount == 0) {
+        Log.warn(`No changes made for item with id ${id} in ${collectionName}`);
+      }
```
it seems that if matchedCount === 1 and modifiedCount === 0  we log the warning but still perform the operation.  is this the intended idea?

peiyaoh replied:
The intended idea is not to throw errors when the document already has the identical values as the updates, which will result in modifiedCount == 0; 

As we discussed in our meeting, an alternative could be added in the future to update the "updatedAt" properties of a document so that the modifiedCount will still be 1.

jadrubio replied:
Yes I rememeber that.  I am not saying we should throw an error merely wondering if we still need to perfom the call to the db to update

peiyaoh replied:
I see. The update should have already been attempted by this point. If we want to prevent the update db update operation, we will first have to check whether the values are the same already, either in the mongo manager or the database manager. Is that what you have in mind?

peiyaoh committed:
Fix formatting in _read function definition

Removed unnecessary space in the _read function assignment for better code consistency.

peiyaoh committed:
Merge pull request #46 from MIACollaborative/fix-space

Fix formatting in _read function definition
